### ISSUE TITLE
Template for R

### ISSUE THREAD

**Issue by c-martinez** [NLeSC/guide#327]
The Python chapter has a reference to our [Python Template](https://github.com/NLeSC/python-template), as a way to start your project with the good practices we encourage everyone to use (license, testing, citation information, etc).

Talking to @PabRod, apparently in R-land, it is more common to use the [usethis](https://github.com/r-lib/usethis) library. Instead of creating a template for R, it would make sense to have a few lines of R that serve the same purpose (start your project with good practices we encourage).

I guess it would look something like this:

```R
library(usethis)
usethis::create_package()
usethis::use_readme_md()
usethis::use_apache_license()
usethis::use_testthat()
usethis::use_github_action()
usethis::use_citation()
usethis::use_cff()
```

Would more experienced R people be able to give their opinion?

**Comment by hechth** [comment]
usethis is really very useful and already does a lot of templetization for you, so you could technically write a small R script to which you pass some parameters, like the name of the package etc. and then just rune the usethis library to init most of the things :)

**Comment by egpbos** [comment]
@maltelueken What do you think about this?

**Comment by maltelueken** [comment]
I agree with the above! `usethis` is also what I would recommend to those who want to start a new R package. Having such a script (maybe publish it as a GitHub Gist?) would be a nice idea.

**Comment by c-martinez** [comment]
During the [research software templates](https://virtual.oxfordabstracts.com/#/event/49081/submission/90) session in RSECon24, there were some comments about "other tools used in R instead of templates":

 - devtools
 - https://github.com/benmarwick/rrtools

Are these equivalent? competitors? I take it our preferred / recommended way is to use `usethis`?

**Comment by PabRod** [comment]
Dear @c-martinez, here my answer:

- In this regard, `devtools` is essentially equivalent to `usethis` (actually, `usethis` is kind of part of `devtools`).
- Never heard of `rrtools`. It is not an official R package (yet).

So yes, `usethis` shall still be our focus.

**Comment by egpbos** [comment]
https://book.the-turing-way.org/reproducible-research/code-reuse/code-reuse-packages#walkthrough-of-creating-a-package-in-r
